response_writer.go

```diff
@@ -30,35 +31,41 @@ type beforeFunc func(ResponseWriter)
 func NewResponseWriter(rw http.ResponseWriter) ResponseWriter {
 	return &responseWriter{
 		ResponseWriter: rw,
-		status:         http.StatusOK,
-		size:           0,
-		beforeFuncs:    nil}
+	}
 }
 
 type responseWriter struct {
 	http.ResponseWriter
 	status      int
 	size        int
 	beforeFuncs []beforeFunc
+	wroteHeader bool
 }
 
 func (rw *responseWriter) WriteHeader(s int) {
 	rw.status = s
 	rw.callBefore()
 	rw.ResponseWriter.WriteHeader(s)
+	rw.wroteHeader = true
 }
 
 func (rw *responseWriter) Write(b []byte) (int, error) {
 	if !rw.Written() {
 		// The status will be StatusOK if WriteHeader has not been called yet
 		rw.WriteHeader(http.StatusOK)
```

Comment:
Turns out we either need to leave this or set `rw.status = http.StatusOK` in order to have `.Status()` return the correct code after calling `Write()` without calling `WriteHeader()` so leaving this for now.